RDKEMW-7130 AAMP Federated Release for Sprint 2508 : Tag 2.8.0

psiva01 opened the pull request.

pstroffolino committed:
VPLAY-10193 pssh parsing for mp4demux (#243)

VPLAY-10193 drm metadata support in mp4demux

Reason for Change: support for parsing drm-specific (pssh, senc, etc.) boxes in mp4demux, and new aampcli 'parse' command
Test Guidance: can use new 'parse' command  to demux an encrypted header/segment and see those boxes processed
Risk: Low (not used by default)

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

nu641001 committed:
VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel (#286)

VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel

Reason for change: Use to SYSTEM clock instead of STEADY clock for comparisons to publish time and manifest end for calculations of live latency, aligned to click time. avoiding bad value that can cause rate correction to wrongly be used.
Risks: Low
Test Procedure: Test with LLD channels
Priority: P1

Signed-off-by: Nandu <nu641001@gmail.com>

jfagunde committed:
VPLAY-10269 Add a script to parse L1 test output (#267)

VPLAY-10269 Add a script to parse L1 test output

Reason for Change: Identify which L1 tests failed or did not complete
Includes help and detection for skipped tests.

Test Procedure: Run the script with L1 tests console output
Risks: Low

gavin-parry committed:
VPLAY-10301 Deprecate use of GST_CHECK_VERSION(1,18,0) (#279)

Reason for change: We no longer wish to support gst < 1.18.0

Test Procedure: Ensure build fails with older gst versions. Also ensure
it builds in bitbake for a target using gst>1.18.0 and in local builds
on Ubuntu.

Risks: Low

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10350 "godspell cleanup, continued"

VPLAY-10350 "godspell cleanup, continued"

Reason for Change: this fixes a small set of bogus spellings, i.e. backwordSerach, briefa, bol sttaus, senario, partiton in core aamp

Test Guidance: no godspell warnings
Risk: Low (mostly just changes in comments)

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10193 replacement for gst_buffer_new_memdup

VPLAY-10193 replacement for gst_buffer_new_memdup

Reason for Change: avoid dependency on gstreamer 1.20
Test Guidance: no more build failure on skyxione-xoe having gst 1.18.5
RIsk: Low (already broken!)

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10265: Revert "VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel (…" (#297)

VPLAY-10265: Revert "VPLAY-10265: Playback Rate Correction Issue Observed in LLD Channel (…" (#297)

This change broke several L2 tests related to trickplay.  Reverting until failure is understood and change reworked.

This reverts commit 65999c961463ba21d45329a6371b1b673ec63111.

dp0000 committed:
rdk-58554 Address the warnings in CPE integration

RDK-58554 Address the warnings in CPE integration

Reason for change: add casting to avoid compilation warning with float vs. integer play rate
Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

kd-syna committed:
VPLAY-10294 Add timestamps to aamp-cli output. (#299)

VPLAY-10249 Add timestamps to aamp-cli output.

Reason for Change:
	Allow the timing of aamp-cli log output to be accurately
	compared to the timing of aamp log output.

Test Guidance:
	Check that aamp-cli log output contains timestamps that are
	consistent with aamp's (in both the values and the formatting).

Risk: Low

Vinish100 committed:
VPLAY-10294 Add timestamps to aamp-cli output (#313)

VPLAY-10294 Add timestamps to aamp-cli output

Reason for change: fix printf format in new aampcli-specific logging macros to avoid osx compilation issus.
Test Procedure: As mentioned in the ticket
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

rajat698 committed:
VPLAY-10344 Incorporate L1 parser script to Github actions CI

VPLAY-10344 Incorporate L1 parser script to Github actions CI

Reason for change: Adds L1 test run script to Github actions L1 tests workflow to PRs of rdkcentral/aamp repo.
 to display pretty utest run

Risks: Low
Priority: P1
---------

Signed-off-by: rajat698 <emailofrajatyadav@gmail.com>

pstroffolino committed:
VPLAY-10294 fix logging for 'list' command (#323)

Reason for Change: fix regression with list command formatting
Test Guidance: use 'list' command in aampcli
Risk: None

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

psiva01 committed:
VPLAY-10434:Crash (37983920-lstring::atof) during ad break of SLE (#322)

VPLAY-10434:Crash (37983920-lstring::atof) during ad break of SLE (#322) 

Reason for change: Replaced assert() with throw std::runtime_error()
exception.
Test Procedure: Refer Jira
Risks: No

Signed-off-by: psiva01 <sivasubramanian.patchaiperumal@ltts.com>

jameslofthouse-RED committed:
VPLAY-10423 Captions stop after advert (discontinuity) (#314)

VPLAY-10423 Captions stop after advert (discontinuity)  

Reason for change: Without PTS restamp enabled, text data is not sent to subtec after an ad break
Test Procedure: See ticket
Risks: Low

Signed-off-by: James Lofthouse <james_lofthouse@comcast.com>
Co-authored-by: James Lofthouse <james_lofthouse@comcast.com>

Vinish100 committed:
VPLAY-10406: Handle zero buffers with HTTP 200 in DE-LL (#324)

VPLAY-10406: Handle zero buffers with HTTP 200 in DE-LL

Reason for change: Make chunk mode injection for LLD configurable
Empty buffers with download successful is now treated as
partial file download to trigger rampdown.
Test Procedure: Check for playback freezes in DE LLD streams
Also validate with L2 1019
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

varshnie committed:
VPLAY-10239:L2 5003 failure (#305)

VPLAY-10239:L2 5003 failure

Reason for change:L2 5003 failure
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: varshnie <varshniblue14@gmail.com>

jfagunde committed:
VPLAY-9399 Improve AAMP logs (#244)

VPLAY-9399 Improve AAMP logs

Reason for change: make it easier to triage various AAMP issues; more consistent use of MIL logging

Test Procedure: Verify AAMP L1 tests pass

Vinish100 committed:
VPLAY-10406: Fix L2 2024 failure due to empty buffer check

Reason for change: Check for zero-byte buffer in successful downloads
Test Procedure: 2024 L2 should pass
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

pstroffolino committed:
VPLAY-10424 unreliable player state eventing when stopping playback

VPLAY-10424 unreliable player state event when stopping playback

Reason for Change: minimally invasive patch to address race condition with player state assignment and events while stopping
- now consistently using idle state when stopping - after flushing events.  This avoids undefined behaviors/race conditions.
- introduces an "isDestructing" parameter so that internally called stop associated with destructing a playerInstance class instance doesn't generate state change events referencing stale player instance

Test Guidance: run l2 tests (especially 1006) and get readout from apps
Risk: Medium

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

* isDestructing param

varshnie committed:
VPLAY-9810:Add droppedFrames to MonitorAV event (#270)

VPLAY-9810:Add droppedFrames to MonitorAV event

Reason for change:Added audio buffered duration as a part of mediaprogress event and dropped frames as part of monitorav event
Risks: Low
Test Procedure: Refer jira ticket
Priority: P2

Signed-off-by: varshnie varshniblue14@gmail.com

pstroffolino committed:
VPLAY-9810 monitor AV typo (#335)

VPLAY-9810 monitor AV typo

Reason for Change: fix typo caught by godspell
Test Guidance: godspell no longer flagging
Risk: None

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
VPLAY-10516: Fix L2 5003 failures in sprint

Reason for change: SetLocalTSBInjection before querying
GetFirstPTS to get the accurate PTS value from TSBReader
Test Procedure: All L2 5003 should pass
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

supriyadappu committed:
VPLAY-10523 : [AAMP] Sprint release version is displayed as 7.06 instead of 7.07 (#336)

VPLAY-10523 : [AAMP] Sprint release version is displayed as 7.06 instead of 7.07 (#336)

Reason for change: Changed AAMP version to 7.07
Test Procedure: Refer Jira
Risks: None

nebutler963 committed:
VPLAY-9840 -AAMP TSB - OOB subtitles – L1 Tests (#237)

VPLAY-9840 -AAMP TSB - OOB subtitles – L1 Tests

Reason for Change: L1 Test for OOB subtitles support when using TSB.
Test Procedure: L1
Risks: Low
---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

jfagunde committed:
VPLAY-10510 Improve L1 test results parsing script (#337)

VPLAY-10510 Improve L1 test results parsing script

Reason for change: Fix wrong results shown in GitHub actions
Summary of changes:
* Update regex patterns to handle complex test names
* Detect timeout failures
* Improve disabled test detection
* Use sets to better performance
* Show specific failure type ([Failed], [Timeout]) in the output
* Fix completed tests incorrectly flagged as incomplete

Test Procedure: Run script with the output of running L1 tests
Risks: Low

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10512 coverity fixes (#333)

VPLAY-10512 coverity fixes

Reason for Change: collection of coverity flagged patches

Test Guidance: general regression

Risk: Low

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
VPLAY-9904: Support PTO for DASH SegmentTimeline format

Reason for change: Handle PTO in SegmentTimeline. Add PTO offset
from timelineStart to skipTime to skip to the right fragment. This also
adjusts for seekMidFragment config. Use a new flag mIsFinalFirstPTS to
determine if we should update PTS in NotifyFirstVideoPTS which will be
the firstPTS of fragment and doesn't include PTO. Added L1.
Test Procedure: Test with streams in VPLAY-10064 and also validate
with the app as in ticket description
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

nu641001 committed:
VPLAY-10590: gst test harness segfault on devices (#364)

VPLAY-10590: gst test harness segfault on devices

Reason for change: Fix crash with gstTestHarness
Risks: Low (tool only fix)
Test Guidance: run gst test harness on devices
Priority: P1

Signed-off-by: nu641001 <nu641001@gmail.com>

pstroffolino committed:
VPLAY-9850  deprecate need for GrowableBuffer::AppendNulTerminator

VPLAY-9850  deprecate need for GrowableBuffer::AppendNulTerminator

Reason for Change: deprecate code that appends NUL characters to GrowableBuffer
Test Guidance: soak in sprint to catch any regressions with hls/ts content playback, including OOB captions with hls ts pts restamping
Risk: Medium

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-9850 l2fix (#367)

VPLAY-9850 deprecate need for GrowableBuffer::AppendNulTerminator

Reason for Change: workaround to avoid l2 test failure

Test Guidance: no l2 test regression

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

* Reason for Change: workaround

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

* Reason for Change: typo

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

---------

Signed-off-by: Stroffolino <philip_stroffolino@cable.comcast.com>

narenr94 committed:
RDKEMW-6727 : Move Plugin-Tests from aamp to middleware utest

RDKEMW-6727 : Move Plugin-Tests from aamp to middleware utest

Reason for Change: Move Plugin-Tests in utests in aamp into middleware similar to RDK-58340
Test Guidance: L1 tests able to run and passing, plus general regression (no functional change expected in core aamp)

Vinish100 committed:
VPLAY-9706: Improve UVE API documentation for local TSB configuration (#374)

VPLAY-9706: Improve UVE API documentation for local TSB configuration

Reason for change: Update UVE documentation with TSB feature details
Test Procedue: Validate documentation is formatted correctly
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

anshephe committed:
VPLAY-8941: Send gstreamer new_segment with applied_rated on all platforms (not just ePLATFORM_AMLOGIC) (#352)

VPLAY-8941: Send gstreamer new_segment with applied_rated on all platforms (not just ePLATFORM_AMLOGIC)

Reason for Change: Rialto video sink has a new property "is-master", to determine if video is master; this is now used by DefaultSocInterface::IsVideoMaster and DefaultSocInterface::IsPlatformSegmentReady.
* Added logging to debug
* Testing using gst_app_src_push_sample
* Don't call NotifyBufferProcessed after sending first buffer on Rialto, even if useWesterosSink is false
* Updated utest fakes and mock
* avoided position reporting issue by limiting change to when Rialto in use

Test Procedure: Tested with VIPA Test App

Risk: Low

Signed-off-by: Andrew Shepherd anshephe@synamedia.com

hthakkar-synamedia committed:
VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

VPLAY-9633 AAMP TSB - Changing audio track LLD results in jump in position

Reason for Change: - Changing the audio track for LLD causes the position to jump to live edge instead of live play position

A bug was discovered in Tunehelper that caused a wrong pts value sent during the flush.

Changes: Clears the mFirstPTS value to trigger the update of the first PTS value sent during the flush
Refactors the code by introducing ReinitializeInjection API

Test Procedure: Refer to the Ticket.

Risks: Low

kd-syna committed:
VPLAY-10558 Coverity fix (#371)

VPLAY-10558 Coverity fix

Reason for Change: Work around a Coverity false-positive by using std::lock_guard instead of explicit lock/unlock

Test Guidance: General regression.

Risk: Low

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-9633 spelling fix to placate godspell

VPLAY-9633 spelling fix to placate godspell
Reason for Change: spelling change in comment to "Re-initializes"

Test Guidance: no godspell warning

Risk: None

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

nu641001 committed:
VPLAY-10430: Enable Mediaprocessor for LLD stream

Reason for change: Enabling media processor for LLD stream for initial flush
Risks: Low
Test Procedure: Test with
Priority: P1

Signed-off-by: nu641001 <nu641001@gmail.com>

cpc005 committed:
VPLAY-10509 [ENT-4082] AAMP TSB - CDAI Ad Presentation freezes on Channel Four

VPLAY-10509 [ENT-4082] AAMP TSB - CDAI Ad Presentation freezes on Channel Four

Reason for Change: When multiple ADs are placed in an ad break then the start position of the Ads is being calculated incorrectly on CH4 due to the structure of the manifest

Test Guidance: Tune to Channel Four with AAMP Managed Local TSB (not FOG) and wait for ad breaks

Risk: Low
---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10509 spelling fix (#381)

Reason for Change: fix introduced typo 'begining'

Test Guidance: no godspell error

Risk: none

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-9850  VPLAY-10697  Revert "VPLAY-9850  deprecate need for GrowableBuffer::AppendNulTerminator" (#365)

VPLAY-9850 VPLAY-10697  Revert "VPLAY-9850  deprecate need for GrowableBuffer::AppendNulTerminator"

Reason for Change: This reverts commit 01b2db19270e26bb9e98f1833702b7a9f57d73a1
VPLAY-10697 "broken L2 2000 testdata7"

Test Guidance: this expected to avoid last few HLS regressions from VPLAY-9850, to be moved back to in development and reworked

Risk: Low

Vinish100 committed:
VPLAY-10694: Add rpath to AAMP binaries (#384)

VPLAY-10694: Add rpath to AAMP binaries

Reason for change: Add RPATH=$ORIGIN to AAMP binaries
for VIPA widget
Test Procedure: Validate DRM playback with VIPA
Risks: Low

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f6eeb33ad2f6f3c3baf50a4663f07788
PRESENT
-->

# Coverity Issue - Data race condition
A wait is performed without a loop. If there is a spurious wakeup, the condition may not be satisfied.

_Medium Impact, CWE-none_
_BAD_CHECK_OF_WAIT_COND_

## How to fix
 Check the wait condition in a loop, with the lock held. The lock must not be released between the condition and the wait.

## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[priv_aamp.cpp:7462](undefined//blob/undefined/priv_aamp.cpp#L7462)

rdkcmf-jenkins commented on streamabstraction.cpp:
```diff
@@ -3389,12 +3395,16 @@ void MediaTrack::OnSinkBufferFull()
+		}
+		
 		// check if cache buffer is full and caching was needed
 		if (IsFragmentCacheFull() && (eTRACK_VIDEO == type) &&
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
fa66964f0e9f006f23d564ed80411b71
NOT_PRESENT
-->

Coverity issue no longer present as of: undefined
<details>
<summary>Show issue</summary>


# Coverity Issue - Data race condition
Accessing "this->cachingCompleted" without holding lock "MediaTrack.mutex". Elsewhere, "MediaTrack.cachingCompleted" is written to with "MediaTrack.mutex" held 3 out of 3 times.

_Medium Impact, CWE-366_
_MISSING_LOCK_



</details>

rdkcmf-jenkins commented on AampEvent.cpp:
```diff
@@ -150,14 +150,16 @@
-		mBufferedDuration(bufferedDuration),
+		mVideoBufferedDurationMs(videoBufferedDuration),
+		mAudioBufferedDurationMs(audioBufferedDuration),
 		mSEITimecode(seiTimecode),
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b847c24d2d0738430234b752556c59dd
PRESENT
-->

# Coverity Issue - Variable copied when it could be moved
"seiTimecode" is passed-by-value as parameter to "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const &)", when it could be moved instead.

_Low Impact, CWE-none_
_COPY_INSTEAD_OF_MOVE_

## How to fix
 Use "std::move(""seiTimecode"")" instead of "seiTimecode".

rdkcmf-jenkins commented on test/gstTestHarness/gst-test.cpp:
```diff
@@ -175,56 +172,59 @@ void GetVideoSegmentPath( char path[MAX_PATH_SIZE], int segmentNumber, VideoReso
+					{
+						gMp4Demux[mediaType] = new Mp4Demux();
+					}
+					gMp4Demux[mediaType]->Parse(ptr,len);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6544e5df56acd0c74e03653967055800
PRESENT
-->

# Coverity Issue - Untrusted loop bound
Passing tainted expression "*this->ptr" to "Parse", which uses it as a loop boundary.

_Medium Impact, CWE-606_
_TAINTED_SCALAR_

## How to fix
 Ensure that tainted values are properly sanitized, by checking that their values are within a permissible range.

psiva01 committed:
Merge branch 'develop' into feature/RDK-58867_federated_2508

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=394) <br/>All committers have signed the CLA.

sayanurag committed:
VPLAY-10785 Anti piracy Watermarking not detected (#409)

VPLAY-10785 Anti piracy Watermarking not detected

Reason for change: The parameter mIsVideonMute is an atomic variable. Directly assigning a value to an atomic variable might have caused this unexpected behaviour. Changing the value of an atomic variable in a thread-safe manner, preventing race conditions fix this issue

Test Procedure: refer test steps in ticket

Risks: Low

Signed-off-by: Anurag Krishnan <akrish513@cable.comcast.com>

Nithya-k13 committed:
VPLAY-10820: Observing Technical fault after waking from Deepsleep (#429)

VPLAY-10820: Observing Technical fault after waking from Deepsleep

Reason for change: Maintained the single secmanager session in middleware component to resolve the closesession failure
Test Procedure: Refer Jira
Risks: Low

Signed-off-by: Nithya-k13 <nithyak936@gmail.com>
